packages/engine/CONTRIBUTING.md

```diff
@@ -0,0 +1,149 @@
+
+To help you help us, we've written this section to give some guidelines on the sort of information that will help us investigate and hopefully help you.
+
+#### Before Submitting A Bug Report
+
+Try and check to see if it's already been reported, the main place to look would be the [Github Issues board](https://github.com/hashintel/hash/issues). If you can't find something similar, then feel free to create a new one! Be sure to label it with the `hengine` label to help the right team find it.
+
+#### How Do I Submit A (Good) Bug Report?
+
+So you've found a bug, next up is collecting useful information to give us to investigate.
+
+* **Give a clear title of what the problem is** (or what you believe it to be)
+* If you can, **describe how to reproduce it**, otherwise try and give as much information about how it happened. (What you did, _how_ you did it, etc.) As an example of some useful information depending on the context:
+  * Runtime parameters
+  * The simulation you were running
+  * The step of the build you were at
+  * Any configuration you've supplied
+* **Describe the environment it was running in**
+  * What hardware, as an example:
+    * Your computer manufacturer
+    * Amount of RAM
+    * CPU model
+  * What operating system was it running on
+  * What versions of the dependencies are you running
+    * e.g. tell us your version of Rust, Python, cmake, etc.
+  * Any environment variables that might have affected it
+* **Provide _all_ relevant logs**
+  * Try not to just provide summarised descriptions of what was logged, we like full log-outputs. If you can reproduce it, try running it with trace logs (RUST_LOG=trace)
+
+### Suggesting Enhancements
+
+If you have ideas of how to improve the hEngine, feel free to submit them on the [Product Feedback section of our community forum](https://community.hash.ai/c/product-feedback/2). If you want a more informal discussion around your ideas feel free to start a conversation on the Discord!
+
+
+[comment]: <> (The following sections are comments until the wishlist section of the new website is live)
+[comment]: <> (#### Before Submitting An Enhancement Suggestion)
+
+[comment]: <> (> TODO)
+
+[comment]: <> (#### How Do I Submit A &#40;Good&#41; Enhancement Suggestion?)
+
+[comment]: <> (> TODO)
+
+### Your First Code Contribution
+
+> TODO
+
+#### Local development
+
+> TODO
+
+### Pull Requests
+
+> WIP - More information incoming regarding PR description formatting, etc. 
+
+#### Some good practices
+
+- Have well encapsulated commits on your PR, this helps with reviewing. If you're comfortable with it, rebasing before opening your PR can help with cleaning up WIP commits.
+- Write unit-tests _as_ you write code. Submitting unit-tests alongside the code within commits improves confidence in your changes and helps us understand your intentions with the code.
+
+
+## Styleguides
```

Comment:
As we want to/do have global policies around style we should publish these styleguides independently, outside of the `packages/engine` folder, and cross-link to them here (and from other package READMEs where language-relevant).

Reply:
```suggestion
## Styleguides

[comment]: <> (Cross-link to global style guides for the repo when they're added)

```

Are we fine with adding it as a comment for now?

Reply:
Yes. Let's leave the Rust one you've written up here anyway in place for the timebeing, until that is ready.